include/deal.II/base/function_parser.h

```diff
@@ -259,10 +256,13 @@ class FunctionParser : public AutoDerivativeFunction<dim>
   FunctionParser(FunctionParser &&) = delete;
 
   /**
-   * Destructor. Explicitly delete the FunctionParser objects (there is one
-   * for each component of the function).
+   * Destructor.
+   *
+   * @note We explicitly declare this destructor here (without the definition)
+   * since we do not expect muParser.h to be available in user projects: i.e.,
+   * the destructor must be defined in the source file.
```

Comment:
It took me twenty seconds to understand what you're trying to say here. I think it's of no importance to the user -- in fact nobody ever has to look up the documentation of a destructor because it takes no arguments and calling it is not a voluntary act.

I would move this comment to the implementation of the destructor to explain why we even have a user-defined destructor and why it's implementation is (presumably) empty.